### ISSUE TITLE
[Questions] Remove ?si=xxx param in shared urls

### ISSUE THREAD

**Issue by ego-lay-atman-bay** [arichorn/uYouPlusExtra#96]
### Have you read the FAQ?

- [X] Yes, I did read the FAQ

### Is there an existing issue/question for this?

- [X] I have searched the existing issues

### Do you think this is a bug?

❌ No, I don't think this is a bug. I will explain below

### My question

I noticed that youtube started adding an ?si=xxx param to the end of urls when sharing a video / playlist. This is most likely tracking info, because it's not necessary. It would be great if there was an option to sanitize links, removing unnecessary url params from shared links.

### Additional context

Here's an example url
https://youtu.be/zrQldlvovhc?si=4WMQXL9YX3jG9kT6

This makes urls longer, and takes more time for users to remove the tracking param, so it would be nice if there was an option to remove these.

**Comment by ego-lay-atman-bay** [comment]
Nice

**Comment by ego-lay-atman-bay** [comment]
That's amazing!

(Also, why does youtube use NO instead of false?)

**Comment by yodaluca23** [comment]
This is kinda related to this https://github.com/qnblackcat/uYouPlus/discussions/1178, if you could do these too that'd be great!

**Comment by yodaluca23** [comment]
Maybe intercept it before it goes to the share sheet and strip it then send it along? I've just realized the severity of this, if we could kind of get this expedited that'd be great!

**Comment by yodaluca23** [comment]
Any progress on this?

**Comment by yodaluca23** [comment]
Sorry to bother, but any progress?

**Comment by yodaluca23** [comment]
> Next release, I will include a new button `YouTimeStamp`
> 
> and not only it copies the url with the timestamp
> 
> it completely removes the tracking! `?si=xxx`
> 
> heres a video of the button and how it works
> 
> 
> 
> https://github.com/arichornlover/uYouEnhanced/assets/78001398/84b35450-c55c-4d08-b02f-b80d1d240e00
> 
> 
> 
> Special Thanks to **bhackel** for helping out with it!
> 
> 

That's cool! But since this is possible, I wonder if can you just hide the share button and then make add in your own share button that would get the link from the YouTube video without the tracking parameter and then send it straight to the iOS style share sheet also incorporating a direct native iOS share sheet.

**Comment by yodaluca23** [comment]
Found [this tweak](https://github.com/jkhsjdhjs/youtube-native-share), that not only removes the "si" URL parameter, but also forces the native iOS share sheet, if you could implement the code from here in two options, one to remove the "si" parameter and one to bypass the YouTube share sheet...

**Comment by yodaluca23** [comment]
Hey, since this commit, https://github.com/arichornlover/uYouEnhanced/commit/59cc39107f419c742bff64722d335e15c4766fe9 the share button just completely doesn't work... I think something was implemented wrong... Also, I couldn't find a switch in settings, is their supposed to be, or is it just not going to be controllable?

**Comment by yodaluca23** [comment]
Yep, can confirm it works now! Thank you so much, this is going to save me so much time manually stripping it! 😃
